### PR TITLE
Replace mkgrub.sh with Python code

### PR DESCRIPTION
**Pull request by agraul**
Fixes https://github.com/cobbler/cobbler/issues/2721

### TODO
- [x] API exposure
- [x] CLI action
- [x] less hardcoding -> config

### CODE REVIEW

**Comment by agraul** [comment]
@SchoolGuy I still got the following error when running the tests:

```
        if choices and default not in choices:
>           raise Exception("field %s default value (%s) is not listed in choices (%s)" % (name, default, str(choices)))
E           Exception: field virt_type default value (auto) is not listed in choices (['xenpv', 'xenfv', 'qemu', 'kvm', 'vmware'])
```

**Comment by nodeg** [comment]
@SchoolGuy Since this PR is ready for review, should it be included in the 3.3.0 release?

**Comment by SchoolGuy** [comment]
@nodeg I would say we do your migrations and then release. I would merge this for V3.3.1

**Comment by SchoolGuy** [comment]
I have decided to delay the V3.3.0 release and want to include this. Thus I have edited the last commit I made. Once I have tried this PR on a VM successfully, I will merge it.

**Comment by SchoolGuy** [comment]
I'll add a littlebit more testing and then when the tests pass we can merge this.

**Comment by SchoolGuy** [comment]
Tests are passing and the coverage is very good. So this should be good now. If we have bugs we will fixup them later. If you don't have the efi modules installed the command fails. This is okay imho but let's see.

**Review by SchoolGuy — APPROVED** [review]
LGTM

File: cobbler/actions/grubimage.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Cobbler's settings. See man(1) grub2-mkimage for available formats.
+"""
+import pathlib
+import subprocess
+import typing
+
+from cobbler import utils
+
+
+# NOTE: does not warrant being a class, but all cobbler actions use a class's
```
Comment (nodeg) [inline cobbler/actions/grubimage.py:13]:
```suggestion
# NOTE: does not warrant being a class, but all Cobbler actions use a class's
```

File: cobbler/actions/grubimage.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+# .run() as the entrypoint
+class GrubImage:
+    """Action to create bootable Grub2 images."""
```
Comment (nodeg) [inline cobbler/actions/grubimage.py:16]:
```suggestion
    """Action to create bootable GRUB 2 images."""
```

File: cobbler/actions/grubimage.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                    self.bootloaders_dir.joinpath(options["binary_name"]),
+                    self.modules + options.get("extra_modules", []),
+                )
+            except subprocess.CalledProcessError:
+                utils.log_exc()
+                # don't create module symlinks if grub2-mkimage is unsuccessful
+                continue
+
+            # Create a symlink for grub2 modules
+            # assumes a single grub can be used to boot all kinds of distros
```
Comment (nodeg) [inline cobbler/actions/grubimage.py:65]:
```suggestion
            # Create a symlink for GRUB 2 modules
            # assumes a single GRUB can be used to boot all kinds of distros
```

File: cobbler/api.py [hunk 4]
```diff
@@ -1844,3 +1844,12 @@ def get_valid_obj_boot_loaders(self, obj):
         :return: Get a list of all valid boot loaders.
         """
         return obj.supported_boot_loaders
+
+    # ==========================================================================
+
+    def create_grub_images(self):
+        """
+        Create the Grub installer images via this API call. It utilizes ``grub2-mkimage`` behind the curtain.
```
Comment (nodeg) [inline cobbler/api.py:1852]:
```suggestion
        Create the GRUB installer images via this API call. It utilizes ``grub2-mkimage`` behind the curtain.
```

File: cobbler/settings/migrations/V3_3_0.py [hunk 5]
```diff
@@ -190,7 +214,18 @@ def migrate(settings: dict) -> dict:
...
+                    'bootloaders_modules': ["btrfs", "ext2", "xfs", "jfs", "reiserfs",
+                                            "all_video", "boot", "cat", "configfile", "echo", "fat", "font", "gfxmenu",
+                                            "gfxterm", "gzio", "halt", "iso9660", "jpeg", "linux", "loadenv", "minicmd",
+                                            "normal", "part_apple", "part_gpt", "part_msdos", "password_pbkdf2", "png",
+                                            "reboot", "search", "search_fs_file", "search_fs_uuid", "search_label",
+                                            "sleep", "test", "true", "video", "mdraid09", "mdraid1x", "lvm", "serial",
+                                            "regexp", "tr", "tftp", "http", "luks", "gcry_rijndael", "gcry_sha1",
+                                            "gcry_sha256"],
+                    'syslinux_dir': "/usr/share/syslinux",
+                    'grub2_mod_dir': "/usr/share/grub2"}
```
Comment (nodeg) [inline cobbler/settings/migrations/V3_3_0.py:228]: This could have been place to the other bootloader entries above to keep the alphabetic ordering.

**Review by nodeg — APPROVED** [review]
I left some suggestions. Otherwise LGTM. Thanks @agraul!

File: cobbler/settings/migrations/V3_3_0.py (hunk 5 above)
Comment (SchoolGuy) [reply cobbler/settings/migrations/V3_3_0.py]: Done with: ba1884d

**Pull request merged**